File: app/src/main/java/com/starsep/sokoban/mvc/GameModel.java
package com.starsep.sokoban.mvc;

import android.support.annotation.NonNull;

import com.starsep.sokoban.gamelogic.HighScore;
import com.starsep.sokoban.gamelogic.Level;
import com.starsep.sokoban.gamelogic.Move;
import com.starsep.sokoban.gamelogic.Position;

public interface GameModel {
	void onPush();

	void onMove();

	void onWin();

	void moveUp();

	void moveDown();

	void moveLeft();

	void moveRight();

	void repeatLevel();

	void undoMove();

	int levelNumber();

	@NonNull
	HighScore stats();

	@NonNull
	Position player();

	@NonNull
	Level level();

	void nextLevel();

	@NonNull
	Move lastMove();

	void onUndoMove();

	void onUndoPush();

	void onSecondElapsed();
}

File: app/src/main/java/com/starsep/sokoban/mvc/GameModel.kt
package com.starsep.sokoban.mvc

import com.starsep.sokoban.gamelogic.HighScore
import com.starsep.sokoban.gamelogic.Level
import com.starsep.sokoban.gamelogic.Move
import com.starsep.sokoban.gamelogic.Position

interface GameModel {
    fun onPush()

    fun onMove()

    fun onWin()

    fun moveUp()

    fun moveDown()

    fun moveLeft()

    fun moveRight()

    fun repeatLevel()

    fun undoMove()

    fun stats(): HighScore

    fun player(): Position

    fun level(): Level

    fun nextLevel()

    fun lastMove(): Move

    fun onUndoMove()

    fun onUndoPush()

    fun onSecondElapsed()
}
